src/Datadog.Trace/Tracer.cs

```diff
@@ -181,9 +183,17 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
                 InitializeLibLogScopeEventSubscriber(_scopeManager, DefaultServiceName, Settings.ServiceVersion, Settings.Environment);
             }
 
-            if (Interlocked.Exchange(ref _firstInitialization, 0) == 1 && Settings.StartupDiagnosticLogEnabled)
+            if (Interlocked.Exchange(ref _firstInitialization, 0) == 1)
             {
-                _ = WriteDiagnosticLog();
+                if (Settings.StartupDiagnosticLogEnabled)
+                {
+                    _ = WriteDiagnosticLog();
+                }
+
+                if (Settings.RuntimeMetricsEnabled)
+                {
+                    _runtimeMetricsWriter = new RuntimeMetricsWriter(Statsd ?? CreateDogStatsdClient(Settings, DefaultServiceName, Settings.DogStatsdPort), 10000);
```

Comment:
I'm not sure if we should be tying the lifetime of the `_runtimeMetricsWriter` object to the specific `Tracer` object, because it's possible (and previously suggested in our docs) to replace the `Tracer.Instance` when wanting to set configuration via code. I'm not sure if we can completely decouple it from the Tracer because of the StatsD configuration, but we should at least make this a static field in the `Tracer` type.

Reply:
You're right. For now I moved it to a static field. Eventually I should add a way to substitute the dogstatsd instance, but I'll take care of it in another PR if that's OK with you